### ISSUE TITLE
Some questions

### ISSUE THREAD

**Issue by xianfangfx** [Zzh-tju/DIoU-SSD-pytorch#16]
In the SSD code, I choose diounms to train with IoU, GIoU, DIoU and CIoU, respectively, and save the result when epoch = 1. The AP evaluation of these results shows that DIoU + diounms is the best, GIoU + diounms is the second, IoU + diounms is the third, and CIoU + diounms is the last. Is this normal? Or what parts of the original code need to be modified?

**Comment by Zzh-tju** [comment]
There is randomness in training, especially in the early stage of training

**Comment by xianfangfx** [comment]
Hello! The gradient of v w.r.t. w and h in the paper are not consistent with the results of my solution. I am very confused. Here is my answer.
the gradient of v w.r.t. w: 8/pi^2 * (arctan(w^gt/h^gt)-arctan(w/h)) * (-h/(w^2+h^2))
the gradient of v w.r.t. h: 8/pi^2 * (arctan(w^gt/h^gt)-arctan(w/h)) * (1/(w^2+h^2))

**Comment by Zzh-tju** [comment]
![ciou](https://user-images.githubusercontent.com/57825730/79187057-d76c4500-7e4d-11ea-8ce5-35666d3981d5.jpg)

**Comment by Zzh-tju** [comment]
the latest loss function has removed the self-defined gradient, since we do not observe any explosion in training.

**Comment by xianfangfx** [comment]
I'm sorry that my previous message was wrong. I've fixed it in the picture below. In addition, is the first picture in paper below from the latest version? Is the second figure in code below need to update the plus and minus symbols?
![微信截图_20200414162902](https://user-images.githubusercontent.com/28485212/79204112-9a647a80-7e6e-11ea-98d1-aa7baa191d8b.png)
![微信截图_20200414163251](https://user-images.githubusercontent.com/28485212/79204125-9d5f6b00-7e6e-11ea-9aa8-273d1f233326.png)

**Comment by Zzh-tju** [comment]
Actually, we are about to remove this paragraph in the latest version of paper. Because even in YOLOv3 (darknet), there is only a minor difference between removing 1/(w²+h²) or not. It only affects simulation experiments.
```
Is the second figure in code below need to update the plus and minus symbols?
```
no

**Comment by xianfangfx** [comment]
Thank you very much for your prompt reply.
I'm sorry I didn't state clearly enough. What I don't understand is that the gradient sign of v w.r.t. w and h in the paper is reversed, while the code is written strictly according to the wrong answer of the old paper. Whether it needs to be modified.

**Comment by Zzh-tju** [comment]
For YOLOv3 and simulation experiments, the answer is not.

**Comment by xianfangfx** [comment]
In box_utils.py, I am confused about the definition of diounms, where center_x2 = (xx1 + xx2) / 2, 
while center_y2 = (yy2 + yy2) / 2, why not center_y2 = (yy1 + yy2) / 2.
![微信截图_20200426100446](https://user-images.githubusercontent.com/28485212/80295478-7aee1b80-87a5-11ea-8b72-d9cbc713250a.png)

**Comment by Zzh-tju** [comment]
you're right. I've fixedd this error a couple days ago. But beta for DIoU-NMS will change and become similar to YOLOv3.

And more improvement for this repo is coming soon. maybe within the next 2 weeks.

**Comment by poppy007** [comment]
![image](https://user-images.githubusercontent.com/33019213/150757064-17fe82a0-212b-44d0-96bb-93621a71646a.png)
I am very confused about the gradient explanation of this place, but the use of this gradient is not reflected in your code, and the author's gradient calculation is wrong, I would like to ask you, what is the function of this gradient here?
